### PR TITLE
Add function to verify publish generated hashes

### PR DESCRIPTION
**Pull request by esclapes**

### CODE REVIEW

File: src/rpc/verifyAccount.js [hunk 1]
```diff
@@ -61,5 +63,14 @@ module.exports = async ({ _id, email, password, productName }) => {
...
+    showOLAB,
   }
 }
+
+function shouldBeShownOLAB(account) {
+  const needsReplyRehashing = !!account && !!account.hashedPasswordReply
+  const needsPublishRehashing = !!account && !!account.hashedPasswordPublish
+
+  // let's keep undefined as the default so we can stop sending this flag
+  return needsPublishRehashing || needsReplyRehashing || undefined
```
Comment (esclapes) [inline src/rpc/verifyAccount.js]: I was not sure about what is the right logic here. I thought that any _imported_ user that get re-hashed after the migration would qualify to see the OLAB page, but I am open to implementing a different logic.

An alternative could be `needsPublishRehashing` and has 2 produclinks, for instance.

Any suggestions @philippemiguet?

File: tests/libs/comparePublishHash.test.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+      const { hash, version } = getVersionAndHash(versionedHash)
+
+      expect(hash).toEqual(expectedHash)
+      expect(version).toEqual(expectedVersion)
+    })
+  })
+
+  describe('good hashes', () => {
+    const goodHashes = [
```
Comment (esclapes) [inline tests/libs/comparePublishHash.test.js]: I only have examples for real v2 hashes (generated locally from known passwords). It would be nice to find `v1` examples too.

File: src/rpc/verifyAccount.js (hunk 1 above)
Comment (philippemiguet) [reply src/rpc/verifyAccount.js]: Hey @esclapes good question. 

From my understanding, only users who have multiple passwords will see the OLAB page. 

So, I think you are right with your alternative, we want to check the number of product links, and if the user has a product hash. If someone has only Publish and gets re-hashed, their password didn't change, so the OLAB page is required for them.

> An alternative could be needsPublishRehashing and has 2 product links, for instance.

**tl;dr; I think that's the right answer ^ 👍**

Comment (esclapes) [reply src/rpc/verifyAccount.js]: Thanks for thinking along with me here, Phil. I will change the code to match the alternative logic. :+1:
